File: Python/Sorting/selection_sort.py
#PROBLEM:
#Sorting Algorithm without the use of in-built Javascript methods

#SOLUTION:

#Helper for function to return the index of the smallest value in an array
def minIndex (arr):
   index = 0

   for i in range(len(arr)):
      if arr[i] < arr[index]: index = i

   return index

def selectionSort (arr):
   sortedArr = []

   for i in range(len(arr)):
      sortedArr.append(arr.pop(minIndex(arr)))
   
   return sortedArr

numbers = [2, 6, 1, 8, 10, 4, 1]

print(selectionSort(numbers))

File: GoLang/Algorithms/sorting/selectionSort.go
package sorting

import "fmt"

func SelectionSort(numbers []int) {
	fmt.Println(selectionSort(numbers))
}

func minIndex(arr []int) int {
	min := 0
	for i:= 0; i < len(arr); i++ {
		if arr[i] < arr[min] {
			min = i
		}
	}

	return min
}

func selectionSort(arr []int) []int {
	sortedArray := []int {}
	arrCopy := arr

	for i:= 0; i < len(arr); i++ {
		sortedArray = append(sortedArray, arrCopy[minIndex(arrCopy)])
		arrCopy = append(arrCopy[:minIndex(arrCopy)], arrCopy[minIndex(arrCopy)+1:]...)
	}

	return sortedArray
}